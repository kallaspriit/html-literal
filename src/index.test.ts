import html from "./index";

describe("html-literal", () => {
  it("should work as intended", async () => {
    const info = {
      name: "Chuck Norris",
      age: 42,
      jokes: [
        "Chuck Norris counted to infinity. Twice.",
        "When the Boogeyman goes to sleep every night he checks his closet for Chuck Norris.",
      ],
      details: {
        Residence: "Navasota, Texas, U.S.",
        Occupation: "Actor, martial artist, film producer, screenwriter, air policeman (U.S. Air Force).",
      },
      weakness: undefined,
      enemy: () => "none",
      getInterests: () => Promise.resolve(),
    };

    const output = html`
      <h1>HTML Tagged Template Literal</h1>
      <p>Name: ${info.name}</p>
      <p>Age: ${info.age}</p>
      <p>Details: <pre>${info.details}</pre></p>
      <p>Weakness: ${info.weakness}</p>
      <p>Enemy: ${info.enemy}</p>
      <p>Interests: ${info.getInterests()}</p>
      <ul>
        ${info.jokes.map(
          joke =>
            html`
              <li>${joke}</li>
            `,
        )}
      </ul>
    `;

    expect(output).toMatchSnapshot();
  });
});
