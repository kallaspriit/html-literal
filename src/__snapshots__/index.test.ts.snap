// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html-literal should work as intended 1`] = `
"<h1>HTML Tagged Template Literal</h1>
      <p>Name: Chuck Norris</p>
      <p>Age: 42</p>
      <p>Details: <pre>[{
  \\"Residence\\": \\"1\\",
  \\"Occupation\\": \\"2\\"
},\\"Navasota, Texas, U.S.\\",\\"Actor, martial artist, film producer, screenwriter, air policeman (U.S. Air Force).\\"]</pre></p>
      <p>Weakness: </p>
      <p>Enemy: [function]</p>
      <p>Interests: [promise]</p>
      <ul>
        <li>Chuck Norris counted to infinity. Twice.</li>
<li>When the Boogeyman goes to sleep every night he checks his closet for Chuck Norris.</li>
      </ul>"
`;
